x-pack/plugins/lens/public/indexpattern_datasource/utils.tsx

```diff
@@ -238,3 +242,181 @@ export function getVisualDefaultsForLayer(layer: IndexPatternLayer) {
     {}
   );
 }
+
+/**
+ * Some utilities to extract queries/filters from specific column types
+ */
+
+/**
+ * Given a Filters column, extract and filter useful queries from it
+ */
+function extractQueriesFromFilters(
+  queries: FiltersIndexPatternColumn['params']['filters'] | undefined
+) {
+  return queries?.map(({ input }) => input).filter(({ query }) => query?.trim() && query !== '*');
+}
+
+/**
+ * Given an Interval column in range mode transform the ranges into KQL queries
+ */
+function extractQueriesFromRanges(column: RangeIndexPatternColumn) {
+  return column.params.ranges
+    .map(({ from, to }) => {
+      let rangeQuery = '';
+      if (from != null && isFinite(from)) {
+        rangeQuery += `${column.sourceField} >= ${from}`;
+      }
+      if (to != null && isFinite(to)) {
+        if (rangeQuery.length) {
+          rangeQuery += ' AND ';
+        }
+        rangeQuery += `${column.sourceField} <= ${to}`;
+      }
+      return {
+        query: rangeQuery,
+        language: 'kuery',
+      };
+    })
+    .filter(({ query }) => query?.trim());
+}
+
+/**
+ * Given an Terms/Top values column transform each entry into a "field: term" KQL query
+ * This works also for multi-terms variant
+ */
+function extractQueriesFromTerms(
+  column: TermsIndexPatternColumn,
+  colId: string,
+  data: NonNullable<FramePublicAPI['activeData']>[string]
+) {
+  const fields = [column.sourceField]
+    .concat(column.params.secondaryFields || [])
+    .filter(Boolean) as string[];
+
+  // extract the filters from the columns of the activeData
+  return data.rows
+    .map(({ [colId]: value }) => {
+      if (value == null) {
+        return;
+      }
+      if (typeof value !== 'string' && Array.isArray(value.keys)) {
+        return {
+          query: value.keys
+            .map((term: string, index: number) => `${fields[index]}: ${term || "''"}`)
```

Comment:
We should escape the values here to avoid the query breaking on terms like `( my term )`